pwiz_tools/Skyline/Controls/Graphs/DetectionPlotData.cs

```diff
@@ -40,6 +42,7 @@ public class DetectionPlotData
         public SrmDocument Document { get; private set; }
         public float QValueCutoff { get; private set; }
         public bool IsValid { get; }
+        public string InvalidReason { get; }

```

Comment:
Not a big fan of this approach. Your constructor should either throw an exception with the reason for failure, or you should separate initialization into two functions: constructor and an "Init()" function which returns the text reason for failing to initialize.

Reply:
Fixed. Used the Init() function instead.